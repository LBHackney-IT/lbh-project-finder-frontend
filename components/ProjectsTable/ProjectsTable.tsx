import Link from "next/link";
import { Project } from "../../types";

const ProjectEntry = (project: Project): React.ReactElement => {
  const { id, projectName, phase, size } = project;
  return (
    <Link href={`/projects/${id}/details`}>
      <tr className="govuk-table__row govuk-table__row--clickable">
        <td className="govuk-table__cell">{projectName}</td>
        <td className="govuk-table__cell">{phase}</td>
        <td className="govuk-table__cell">{size}</td>
      </tr>
    </Link>
  );
};

const ProjectsTable = ({
  projects,
}: {
  projects: Project[];
}): React.ReactElement => (
  <table className="govuk-table">
    <thead className="govuk-table__head">
      <tr className="govuk-table__row">
        <th scope="col" className="govuk-table__header">
          Project Name
        </th>
        <th scope="col" className="govuk-table__header">
          Phase
        </th>
        <th scope="col" className="govuk-table__header">
          Size
        </th>
      </tr>
    </thead>
    <tbody className="govuk-table__body">
      {projects.map((result) => (
        <ProjectEntry key={result.id} {...result} />
      ))}
    </tbody>
  </table>
);

export default ProjectsTable;
