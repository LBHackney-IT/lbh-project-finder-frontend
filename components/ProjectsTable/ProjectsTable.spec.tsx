import { render } from '@testing-library/react';
import ProjectsTable from './ProjectsTable';

describe('ProjectsTable component', () => {
    const props = [
        { project_id: 1, project_name: "something", stage: "discovery", size: "big", type: "tech" }
    ]
    it('should render a list of projects', () => {
        const { asFragment } = render(<ProjectsTable projects={props} />);
        expect(asFragment()).toMatchSnapshot();
    })
});