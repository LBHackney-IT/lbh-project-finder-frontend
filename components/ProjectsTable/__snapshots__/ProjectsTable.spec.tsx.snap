// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectsTable component should render a list of projects 1`] = `
<DocumentFragment>
  <table
    class="govuk-table lbh-table"
  >
    <thead
      class="govuk-table__head"
    >
      <tr
        class="govuk-table__row"
      >
        <th
          class="govuk-table__header"
          scope="col"
        >
          Project Name
        </th>
        <th
          class="govuk-table__header"
          scope="col"
        >
          Stage
        </th>
        <th
          class="govuk-table__header"
          scope="col"
        >
          Size
        </th>
        <th
          class="govuk-table__header"
          scope="col"
        >
          Project Type
        </th>
      </tr>
    </thead>
    <tbody
      class="govuk-table__body"
    >
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell"
        >
          Test Project
        </td>
        <td
          class="govuk-table__cell"
        >
          Discovery
        </td>
        <td
          class="govuk-table__cell"
        >
          Large
        </td>
        <td
          class="govuk-table__cell"
        >
          Tech
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
